operator/controllers/btpoperator_controller.go

```diff
@@ -142,7 +143,7 @@ func (r *BtpOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if !cr.ObjectMeta.DeletionTimestamp.IsZero() && cr.Status.State != types.StateDeleting {
-		return ctrl.Result{}, r.UpdateBtpOperatorState(ctx, cr, types.StateDeleting)
+		return ctrl.Result{}, r.UpdateBtpOperatorStatusAndLogStateChange(ctx, cr, types.StateDeleting, HardDeleting, "BtpOperator is to be deleted")
```

Comment:
I see the benefit of placing a logger here, but I'm not sure about logging condition change

Reply:
Logging removed